lib/process/watch.go

```diff
@@ -101,16 +104,19 @@ func (p *Process) watchCertificate(ctx context.Context, client *kubernetes.Clien
 }
 
 // startAuthGatewayWatch launches watcher that monitors config map with
-// auth gateway configuration.
+// auth gateway configuration and updates Teleport configuration
+// appropriately.
 func (p *Process) startAuthGatewayWatch(ctx context.Context, client *kubernetes.Clientset) error {
 	go func() {
+		ticker := time.NewTicker(time.Second)
+		defer ticker.Stop()
 		for {
 			err := p.watchAuthGateway(ctx, client)
```

Comment:
From what I can tell, `watchAuthGateway` is also a loop exiting only when `ctx` has expired. Why no have it run inside of this one - or why is this loop required if you can do the same inside of `watchAuthGateway` (but instead of returning on error - you can loop forever until context expiration?

Reply:
Because IIRC watch can expire so this loop is used to restart the watch loop. Certificate update (in the same file) is implemented the same way.